pkg/core/blockchain.go

```diff
@@ -1553,11 +1554,16 @@ func (bc *Blockchain) verifyTxWitnesses(t *transaction.Transaction, block *block
 		return fmt.Errorf("%w: %d vs %d", ErrTxInvalidWitnessNum, len(t.Signers), len(t.Scripts))
 	}
 	interopCtx := bc.newInteropContext(trigger.Verification, bc.dao, block, t)
+	gasLimit := t.NetworkFee
 	for i := range t.Signers {
-		err := bc.verifyHashAgainstScript(t.Signers[i].Account, &t.Scripts[i], interopCtx, t.NetworkFee)
+		if gasLimit < 0 {
+			return fmt.Errorf("witness #%d: insufficient network fee", i)
```

Comment:
What's the scenario that could possibly trigger this check? Insufficient GAS problem should be handled by `verifyHashAgainstScript` already.

We need a test for this problem (network fee not paying enough for execution of all scripts) anyway though.

Reply:
That was a redundant check, so I dropped it. Also, test is added.